library/public/image.h

```diff
@@ -27,33 +28,80 @@ class F3D_EXPORT image
     }
   };
 
+  /**
+   * Create an empty image
+   */
   image();
+
+  /**
+   * Create an image from file, the following formats are supported:
+   * PNG, PNM, TIFF, BMP, HDR, JPEG, GESigna, MetaImage, TGA.
+   * Can throw an image::exception in case of failure.
+   */
   explicit image(const std::string& path);
 
+  /**
+   * Image destructor
+   */
   ~image();
 
+  //@{
+  /**
+   * Copy/move constructors/operators
+   */
   image(const image& img);
   image& operator=(const image& img);
-
   image(image&& img);
   image& operator=(image&& img);
+  //@}
+
+  //@{
+  /**
+   * Comparison operators, uses image::compare
+   */
+  bool operator==(const image& reference) const;
+  bool operator!=(const image& reference) const;
+  //@}
 
+  //@{
+  /**
+   * Set/Get image resolution
+   */
   unsigned int getWidth() const;
   unsigned int getHeight() const;
-
   image& setResolution(unsigned int width, unsigned int height);
+  //@}
 
+  //@{
+  /**
+   * Set/Get image channel count
+   */
   unsigned int getChannelCount() const;
   image& setChannelCount(unsigned int dim);
-
+  //@}
+
+  //@{
+  /**
+   * Set/Get image buffer data
+   * Its size is expected to be width * height * channelCount
+   *
+   */
   image& setData(unsigned char* buffer);
   unsigned char* getData() const;
-
-  bool compare(const image& reference, image& result, double threshold, double& error) const;
-
-  bool operator==(const image& reference) const;
-  bool operator!=(const image& reference) const;
-
+  //@}
+
+  /**
+   * Compare current image to a reference using the provided threshold.
+   * If the comparison fails, output the resulting diff and error and return false,
+   * return true otherwise.
+   * Can throw an image::exception if images are not sane.
+   */
+  bool compare(const image& reference, image& diff, double threshold, double& error) const;
```

Comment:
@Meakk should this be ` compare(const image& reference, double threshold, image& diff, double& error) const;` instead ?